### ISSUE TITLE
Handling lots of upload files

### ISSUE THREAD

**Issue by Wouter0100** [adopted-ember-addons/ember-file-upload#81]
Firstly, thanks for this library! Seems to be really easy to use so far, although I am currently struggling with uploading lots of files. I hope you're able to help me further.

I'm trying to upload ~1 to >4000 files (images) using this library. When selecting ~100 files, it works fine, although after ~500 files the initial load takes a while and files are processed slowly, very slowly. 

After doing some benchmarks and performance tests, I came access this screen:
![image](https://user-images.githubusercontent.com/864520/33084131-73ef8294-cee1-11e7-8001-23d0715b4b3c.png)

It seems to do with Ember's Observable pattern, although I'm not sure how to tackle this issue. Any suggestions/ideas? The code is nearly the same as the example. 

View
```html
{{#file-dropzone name=(concat "upload-" model.id) as |dropzone queue|}}
    <div class="modal-header">
      {{t
        'project.import.upload-modal.title'
      }}
    </div>
    <div class="modal-body">
      <p>
        {{#if dropzone.active}}
          {{#if dropzone.valid}}
            Drop to upload
          {{else}}
            Invalid
          {{/if}}
        {{else}}
          {{#if dropzone.supported}}
            Drag and drop images onto this area to upload them or
          {{/if}}

          {{#file-upload name=(concat "upload-" model.id)
                         accept="image/*"
                         multiple=true
                         onfileadd=(action "uploadImage")}}
            <a id="upload-image" tabindex=0>add an image.</a>
          {{/file-upload}}
        {{/if}}
      </p>

      <p>Uploading {{queue.files.length}} files ({{queue.progress}}%). Succesfully uploaded {{uploaded}} files, {{duplicates}} duplicates and {{failed}} files failed.</p>
    </div>

    <div class="modal-footer">
      <button class="btn btn-default" {{action 'cancel'}}>
        {{t 'project.import.upload-modal.cancel'}}
      </button>
    </div>
  {{/file-dropzone}}
```

Controller
```js
  uploadPhoto: task(function * (file) {
    let response = yield file.upload(this.get('url'));

    if (response.body.processed) {
      this.get('refreshModel').perform();
      this.get('queue').get('uploaded').addObject(file);
    } else {
      if (response.body.duplicates) {
        this.get('queue').get('duplicates').addObject(file);
      } else {
        this.get('queue').get('failed').addObject(file);
      }
    }
  }).maxConcurrency(2).enqueue(),
```

Any help would be really appreciated. Thanks!

**Comment by tim-evans** [comment]
Huh. Will need to look into this. String manipulation here seems to be a biggie. Hrrrm.

**Comment by Wouter0100** [comment]
Hmm, after you mentioned string manipulation I did some more testing. It turns out `{{queue.progress}}` is really heavy. When removing this, I got ~1200 files uploaded without any hassle. ~3600 becomes "laggy". But hey, that's already a hell of an improvement :+1: 

Currently testing some more with my own code and some other improvements. In the end I hope to get ~14000 files uploaded, but I guess with these many files I also hit some browser related limitations :+1:
